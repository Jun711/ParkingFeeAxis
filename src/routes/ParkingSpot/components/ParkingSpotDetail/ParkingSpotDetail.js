import React, { Component } from 'react'
import { Text, ScrollView, View } from 'react-native'
import styles from './ParkingSpotDetailStyles'
import IconWithText from '..././../../components/IconWithText/IconWithText'

export default class ParkingSpotDetail extends Component {

  constructor(props) {
    super(props)
    this._presentRate = this.props.presentRate ? this.props.presentRate : 1
    this._hourCount = this.props.hourLimit && this.props.hourLimit > 1 ? 'hours' : 'hour'
  }

  render() {
    return (
      <ScrollView style={styles.container}>
        <View>
          <IconWithText
            icon='money'
            message={`It is $${this._presentRate} per hour.`}
          />
          <IconWithText
            icon='hourglass-start'
            message={`Time limit is ${this._presentRate} ${this._hourCount}.`}
          />
        </View>
        <Text
          allowFontScaling={true}
          style={styles.detailText}>{this.props.calloutDetail}</Text>
      </ScrollView>
    )
  }
}
