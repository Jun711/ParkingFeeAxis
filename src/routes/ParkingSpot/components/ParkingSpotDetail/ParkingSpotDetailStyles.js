import { StyleSheet } from 'react-native'
import { SECONDARY_FONT_SIZE, TEXT_COLOR } from '../../../../util/constants'

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  detailText: {
    fontSize: SECONDARY_FONT_SIZE,
    paddingLeft: 5,
    color: TEXT_COLOR,
  }
})

export default styles