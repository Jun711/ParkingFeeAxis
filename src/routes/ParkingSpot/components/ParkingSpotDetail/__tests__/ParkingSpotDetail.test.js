import 'react-native'
import React from 'react'
import ParkingSpotDetail from '../ParkingSpotDetail'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'

describe('<ParkingSpotDetail />', () => {
  it('ParkingSpotDetail renders correctly', () => {
    const parkingSpotDetailComponent = renderer.create(
      <ParkingSpotDetail/>
    ).toJSON()
    expect(parkingSpotDetailComponent).toMatchSnapshot()
  })
})

