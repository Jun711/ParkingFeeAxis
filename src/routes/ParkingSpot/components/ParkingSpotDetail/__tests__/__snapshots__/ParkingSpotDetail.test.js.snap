// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ParkingSpotDetail /> ParkingSpotDetail renders correctly 1`] = `
<RCTScrollView
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View>
    <View>
      <View
        style={
          Object {
            "flex": 4,
            "margin": 10,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "flex-start",
              "paddingLeft": 5,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 12,
                  },
                  Object {
                    "color": "#38414e",
                    "fontSize": 18,
                  },
                  Object {
                    "fontFamily": "FontAwesome",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              
            </Text>
          </View>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#000",
                "flex": 8,
                "fontSize": 18,
                "textAlign": "left",
              }
            }
          >
            It is $1 per hour.
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 4,
            "margin": 10,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "flex-start",
              "paddingLeft": 5,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 12,
                  },
                  Object {
                    "color": "#38414e",
                    "fontSize": 18,
                  },
                  Object {
                    "fontFamily": "FontAwesome",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              
            </Text>
          </View>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#000",
                "flex": 8,
                "fontSize": 18,
                "textAlign": "left",
              }
            }
          >
            Time limit is 1 hour.
          </Text>
        </View>
      </View>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#000",
          "fontSize": 14,
          "paddingLeft": 5,
        }
      }
    />
  </View>
</RCTScrollView>
`;
